### PR TITLE
S3010, S2696: Add support for deconstruction

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**

### CODE REVIEW

**Comment by martin-strecker-sonarsource** [comment]
The indentation was off in one file. Make sure to use the [*Hide whitespace* option of GitHub](https://cloudfour.com/thinks/quick-tip-how-to-hide-whitespace-changes-in-git-diffs/#github).

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM, there's some cleanup that can be done before merging

**Comment by pavel-mikula-sonarsource** [comment]
There are unnecessary usings

**Comment by pavel-mikula-sonarsource** [comment]
Changes after my approval were too big. I'm asking @andrei-epure-sonarsource to take one more look next week.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFrom.cs [hunk 1]
```diff
@@ -97,19 +99,24 @@ public abstract class StaticFieldWrittenFrom : SonarDiagnosticAnalyzer
...
+        private static void CollectLocationOfStaticField(SemanticModel semanticModel, MultiValueDictionary<IFieldSymbol, Location> locationsForFields, ExpressionSyntax expression)
         {
-            if (context.SemanticModel.GetSymbolInfo(expression) is { Symbol: IFieldSymbol { IsStatic: true } fieldSymbol })
+            if (StaticFieldSymbol(semanticModel, expression) is { } fieldSymbol)
             {
                 locationsForFields.Add(fieldSymbol, expression.GetLocation());
             }
         }
+
+        private static IFieldSymbol StaticFieldSymbol(SemanticModel semanticModel, SyntaxNode node) =>
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFrom.cs:114]: could you please add the `Get` prefix here, to underline it's getting it from the SemanticModel for the node?

see https://sonarsource.github.io/rspec/#/rspec/S2365 - when I see "StaticFieldSymbol" I generally think that it should be a cheap call (it's a property-like name), however this invokes the semantic model so it's not cheap

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM, minor nitpick on method name, I'll merge this as is because it's been staying out for long and you have many things ongoing.

IMO you can make the rename on a separate PR that won't be blocking anything and can be merged whenever

**Pull request merged**